src/libYARP_sig/src/yarp/sig/PointCloudUtils-inl.h

```diff
@@ -67,31 +67,51 @@ inline void copyColorData(yarp::sig::PointCloud<T1>& pointCloud,
-                                                         const yarp::sig::IntrinsicParams& intrinsic)
+                                                         const yarp::sig::IntrinsicParams& intrinsic,
+                                                         const yarp::sig::utils::OrganizationType organizationType)
 {
     yAssert(depth.width()  != 0);
     yAssert(depth.height() != 0);
     yAssert(depth.width()  == color.width());
     yAssert(depth.height() == color.height());
     size_t w = depth.width();
     size_t h = depth.height();
+    size_t pointNumber = 0;
     yarp::sig::PointCloud<T1> pointCloud;
-    pointCloud.resize(w, h);
-
+    if (organizationType == yarp::sig::utils::OrganizationType::Organized){
+        pointCloud.resize(w, h);
+    }
     for (size_t u = 0; u < w; ++u) {
         for (size_t v = 0; v < h; ++v) {
-            // Depth
-            // De-projection equation (pinhole model):
-            //                          x = (u - ppx)/ fx * z
-            //                          y = (v - ppy)/ fy * z
-            //                          z = z
-            pointCloud(u,v).x = (u - intrinsic.principalPointX)/intrinsic.focalLengthX*depth.pixel(u,v);
-            pointCloud(u,v).y = (v - intrinsic.principalPointY)/intrinsic.focalLengthY*depth.pixel(u,v);
-            pointCloud(u,v).z = depth.pixel(u,v);
+            if (organizationType == yarp::sig::utils::OrganizationType::Organized){
+                // Depth
+                // De-projection equation (pinhole model):
+                //                          x = (u - ppx)/ fx * z
+                //                          y = (v - ppy)/ fy * z
+                //                          z = z
+                pointCloud(u,v).x = (u - intrinsic.principalPointX)/intrinsic.focalLengthX*depth.pixel(u,v);
+                pointCloud(u,v).y = (v - intrinsic.principalPointY)/intrinsic.focalLengthY*depth.pixel(u,v);
+                pointCloud(u,v).z = depth.pixel(u,v);
+                copyColorData(pointCloud, color, u, v);
 
-            copyColorData(pointCloud, color, u, v);
+            } else if (organizationType == yarp::sig::utils::OrganizationType::Unorganized) {
+                if (depth.pixel(u,v) > 0){
+                    T1 point;
+                    point.x = (u - intrinsic.principalPointX)/intrinsic.focalLengthX*depth.pixel(u,v);
+                    point.y = (v - intrinsic.principalPointY)/intrinsic.focalLengthY*depth.pixel(u,v);
+                    point.z = depth.pixel(u,v);
+                    point.r = color.pixel(u,v).r;
+                    point.g = color.pixel(u,v).g;
+                    point.b = color.pixel(u,v).b;
+                    pointCloud.push_back(point);
+                    pointNumber ++;
+                }
+            }
         }
     }
+    if  (organizationType == yarp::sig::utils::OrganizationType::Unorganized) {
+        pointCloud.resize(pointNumber);
```

Comment:
Is this actually necessary, along with the `pointNumber` variable? The cloud should already have exactly `pointNumber` elements after the last `push_back` call inside the loop.

Reply:
@PeterBowman you anticipated me 😆

Reply:
I can try delete it, I think maybe is not necessary

Reply:
updated

Reply:
@ste93 the `pointNumber` variable is now unused.